docs/csharp/language-reference/builtin-types/struct.md

```diff
@@ -0,0 +1,56 @@
+---
+title: "Structure types - C# reference"
+ms.date: 02/20/2020
+f1_keywords: 
+  - "struct_CSharpKeyword"
+helpviewer_keywords: 
+  - "struct keyword [C#]"
+  - "structs [C#], struct keyword"
+ms.assetid: ff3dd9b7-dc93-4720-8855-ef5558f65c7c
+---
+# Structure types (C# reference)
+
+A *structure type* (or *struct type*) is a [value type](value-types.md) that encapsulates data and related functionality. As the following example shows, you use the `struct` keyword to define a structure type:
+
+[!code-csharp[struct example](~/samples/csharp/language-reference/builtin-types/StructType.cs#StructExample)]
+
+Because a structure type is a value type, a variable of a structure type contains an instance of the type. That is called value semantics. By default, variables are copied on assignment, passing an argument to a method, or returning a method result. In the case of a structure-type variable, an instance of the type is copied (for the examples, see [Value types](value-types.md)). In high-performance scenarios that involve large structure types, you can avoid value copying and pass around an alias to a value-type variable. For information about how to do that, see [Write safe and efficient C# code](../../write-safe-efficient-code.md).
```

Comment:
@BillWagner what do you think of the "alias to a value-type variable" wording?

I've noticed that in the docs the preferred wording is "by reference". However I'm afraid that it might be confusing and misleading. How?

Imagine the following code:

```csharp
int b = 0;
Change(ref b);
Console.WriteLine(b);

void Change(ref int a)
{
    a = 4;
}
```

The above code would print `4`. When we say that we pass `int` by reference, we might imply that there is reference semantics here, that *if `int` were a reference type*, the above code would be equivalent to the following (you wouldn't need `ref`):

```csharp
int b = 0;
Change(b);
Console.WriteLine(b);

void Change(int a)
{
    a = 4;
}
```

And that's not true. Because what is passed is not the reference to a variable but an alias to a variable. That's why whatever is done with that alias is also reflected on the "original" variable. Personally, thinking in aliases instead of references helped me a lot, especially with ref locals and ref returns.

However, if you think that it's not a good idea to deviate from the current docs language, I'll use "by reference" wording in this article as well.

Reply:
> When we say that we pass `int` by reference, we might imply that there is reference semantics here, that _if `int` were a reference type_...

This is how I'm demonstrating the confusion that the [note in the `ref` article](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/ref#passing-an-argument-by-reference) warns about :).

Reply:
I looked at the discussion on #4561, and "by reference:" would be preferred.

In time, I'd like to follow more of the recommendations Eric posted in https://github.com/dotnet/docs/issues/4561#issuecomment-371319556. For now though, let's keep the current language.

Reply:
@BillWagner I'm confused; by the "current language" you mean

1. the current language of the article, that is, "pass around an alias to a value-type variable" OR
1. the current language of the docs, that is, I should rephrase the article to use "by reference"?

I guess, you mean variant 2. Is that correct?

Reply:
Sorry about that. Number 2 is correct. Thanks for asking.